### DISCUSSION TITLE
LocalForage Browser Viewer

### DISCUSSION THREAD

**Discussion by GH20223** [capacitor-community/sqlite#334] [Q&A] [answered]
Hi,

I've managed to create a simple CRUD app for web and ios with angular/capacitor and this great plugin.

For iOS database debugging I am able to read the sqlite file with a tool like **DB BROWSER FOR SQLITE**. 
But what can I use for browser debugging?
In my browser it looks like this:
![image](https://user-images.githubusercontent.com/115704197/195871725-48d3cfc4-ff1b-4bc9-9a9b-b92ecfca7e4d.png)

I've tried the extension LocalForage Explorer but without success.
What else could you try?

Thanks in any case for your help.

**Comment by jepiqueau** [comment, marked as answer]
@GH20223 There is no way to read the database from the local forage it is store as a Uint8Array as from the browser you cannot write the file on your file system. i know some have done it. You can make your own page in your app to look at the database schema, get the table list and query the data for each table

**Comment by jepiqueau** [comment]
@GH20223 Two methods have been added to the Web plugin `getFromLocalDiskToStore` and `saveToLocalDisk` so you will be able to open the database saved to your local disk with DB Browser for SQLite
